compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs

```diff
@@ -238,7 +240,9 @@ impl<'tcx> ConstToPat<'tcx> {
                 }
             } else if !self.saw_const_match_lint.get() {
                 match cv.ty().kind() {
-                    ty::RawPtr(pointee) if pointee.ty.is_sized(self.tcx(), self.param_env) => {}
+                    ty::RawPtr(..) if have_valtree => {
+                        // This is a good raw pointer, it was accepted by valtree construction.
+                    }
                     ty::FnPtr(..) | ty::RawPtr(..) => {
```

Comment:
The actual logic we want here might be "if we don't have a valtree and there was no structural match violation detected with the type-based traversal, then it must be a raw pointer thing so show the lint". That might fix https://github.com/rust-lang/rust/issues/116929. But I'm not sure if pointers are the only case where we fail to build a valtree but do not detect a structural match violation.